crates/bevy_ecs/src/event/base.rs

```diff
@@ -421,3 +424,28 @@ pub(crate) struct EventInstance<E: BufferedEvent> {
     pub event_id: EventId<E>,
     pub event: E,
 }
+
+/// # Warning
+///
+/// This struct should only be instantiated internally to this crate.
+///
+/// [`EventKey`]s are created in:
+/// - `crate::observer::trigger_dynamic_ref_with_caller`
+/// - `crate::observer::trigger_dynamic_ref_with_caller`
+/// - `crate::observer::observer_multiple_events`
+/// - `crate::observer::observer_dynamic_trigger`
+/// - `crate::observer::runner::hook_on_add`
+#[derive(Debug, Copy, Clone, Hash, Ord, PartialOrd, Eq, PartialEq)]
+pub struct EventKey(pub(crate) ComponentId);
+
+impl EventKey {
+    /// Returns id of the underlying [`Event`].
+    /// Used internally in:
```

Comment:
This list will be really hard to maintain. Better to leave this off, and allow users to use their IDEs go-to-uses instead.